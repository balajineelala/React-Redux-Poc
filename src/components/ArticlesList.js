import React from "react";
import ArticleDetails from "./ArticleDetails";
import { useRouteMatch, Link, Route } from "react-router-dom";
import PropTypes from "prop-types";

function ArticlesList({ articles }) {
  let { url } = useRouteMatch();

  if (articles.length < 1) {
    return null;
  }

  return (
    <div>
      {articles.map((article) => {
        return (
          <li key={article.id}>
            <Link to={`${url}/${article.id}`}>{article.title}</Link>
            <Route path={`${url}/${article.id}`} component={ArticleDetails} />
          </li>
        );
      })}
    </div>
  );
}

ArticlesList.propTypes = {
  articles: PropTypes.array,
};

export default React.memo(ArticlesList);
