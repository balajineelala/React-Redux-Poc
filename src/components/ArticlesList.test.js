import React from "react";
import { shallow } from "enzyme";
import ArticlesList from "./ArticlesList";
import { checkPropTypes } from "prop-types";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"), // use actual for all non-hook parts
  useParams: () => ({
    articles: "articles",
    id: "id1",
  }),
  useRouteMatch: () => ({ url: "/articles/id1" }),
}));

describe("ArticlesList", () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      articles: [],
    };
    wrapper = shallow(<ArticlesList {...props} />);
  });

  it("render component without errors", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("render having props", () => {
    props = {
      articles: [
        {
          id: "test id",
          title: "test title",
          body: "test body",
          userId: "test userId",
        },
      ],
    };
    const component = shallow(<ArticlesList {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("render having no data", () => {
    props = {
      articles: [],
    };
    const component = shallow(<ArticlesList {...props} />);
    expect(component).toMatchSnapshot();
  });

  //checking for PropTypes
  describe("checking PropTypes", () => {
    it(" should not throw a warning", () => {
      const expectedProps = {
        articles: [],
      };

      const propsError = checkPropTypes(
        ArticlesList.propTypes,
        expectedProps,
        "props",
        ArticlesList.name
      );
      expect(propsError).toBeUndefined();
    });
  });
});
