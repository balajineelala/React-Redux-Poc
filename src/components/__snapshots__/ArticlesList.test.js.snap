// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList render component without errors 1`] = `ShallowWrapper {}`;

exports[`ArticlesList render having no data 1`] = `ShallowWrapper {}`;

exports[`ArticlesList render having props 1`] = `ShallowWrapper {}`;
